### ISSUE TITLE
make?

### ISSUE THREAD

**Issue by texrg** [simoncozens/libtexpdf#3]
Very interesting project, but what compiling it? Is possible to create README or other way?

**Comment by simoncozens** [comment]
I certainly need to write documentation for this. To compile it, run:

```
aclocal
automake --force-missing --add-missing
autoreconf
./configure
make
```

**Comment by texrg** [comment]
what is 'alocal' ?

on mac
$ automake --force-missing --add-missing
configure.ac: error: no proper invocation of AM_INIT_AUTOMAKE was found.
configure.ac: You should verify that configure.ac invokes AM_INIT_AUTOMAKE,
configure.ac: that aclocal.m4 is present in the top-level directory,
configure.ac: and that aclocal.m4 was recently regenerated (using aclocal)
Makefile.am: installing './INSTALL'
Makefile.am: error: required file './NEWS' not found
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './AUTHORS' not found
Makefile.am: error: required file './ChangeLog' not found
configure.ac:18: error: required file 'config.h.in' not found
Makefile.am:3: error: Libtool library used but 'LIBTOOL' is undefined
Makefile.am:3:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'

on linux all instruction working ... and ?
no example? any C file? How create a simple pdf? any test? 
make test
make example
?

I would like to create document from C (similar sile but from C, now with all features)
Where is API only simple draw line (from dvi) draw font glyph.

**Comment by alerque** [comment]
As best I can make out the current from-scratch way to build this independently looks like this (or will when #13 is merged):

```sh
autoreconf --install
./configure
make
```

`aclocal` *may* still be needed on non-linux systems, I'm not sure.

Eventually this probably needs to get updated so that releases with `make dist` include the right stuff to skip straight to `./configure`

**Comment by snan** [comment]
Also people should know that if you're compiling sile, this gets pulled down as a submodule and compiled as part of that process. (I misunderstood that so I spent some time making a separate .deb out of libtexpdf at first, and I bungled the linking so that didn't even work. I ended up just removing it and relying on the recursive pull in the sile repo.)

**Comment by alerque** [comment]
@snan Right, packaging this separately from SILE is not easy. It can be done, but right now we kind of just assume it is bundled as part of the SILE build.

By the way I'm curious what you were building `.deb` packages for. I have done DEB packages (targeting Ubuntu) already for SILE and have them hosted in a PPA, but would be really interested in hearing if you are working on official Debian packaging or something like that! I would be happy to facilitate that undertaking.

**Comment by snan** [comment]
I was only making .debs for my own use with FPM 💁🏻‍♀️  
I'm not a Debian maintainer and it seems to me that making debs is really difficult but FPM has been a godsend.

I wanted to make three debs, one for libtexpdf, one for the luarocks that aren't already in Debian, and one for Sile itself.

I bungled the first two! Libtexpdf I must've borked some linking, and luarocks ended up hardcoding paths into the temporarily tree I built them in at first. I ended up luarocksing straight onto /usr/local after all 💔 which is what I wanted to avoid doing in the first place. I did at least manage to trim the rockspec file to only the ones that weren't already available in Debian so I could --only-deps the missing ones. (Also it got compiled with lua5.1 even though I also have 5.4 installed because I couldn't figure that part out either. I ain't the sharpest tool in the proverbial when it comes to computer stuff... but I'm really digging sile, what an amazing typesetting program!)
